feat: add docs for running with Postgres (#175)
openfga/openfga.dev: docs/content/getting-started/setup-openfga.mdx

### Configuring Data Storage

OpenFGA supports multiple storage engine options including:

- `memory` - The memory storage engine is the default, but it is not persistent (data is lost between server restarts).
- `postgres` - The Postgres storage engine requires a Postgres instance that the OpenFGA server can reach.

#### Postgres

> Let's assume you want to run OpenFGA and Postgres in containers. The first thing we should do is create a new network as this will make communication between containers a bit simplier:

Reviewer: ```suggestion
Let's assume you want to run OpenFGA and Postgres in containers. The first thing we should do is to create a new network as this will make communication between containers a bit simpler:
```

Author: Thanks!

```console
docker network create openfga
```

Start Postgres in the network you created above:
```console

> docker run -d --name postgres --network=openfga -p 5432:5432 -e POSTGRES_USER=postgres -e POSTGRES_PASSWORD=password postgres:14

Reviewer 2: If they are on the same network, you do not need to expose port 5432 to the host

Author: Good point. And I also wasn't exposing ports in the final command. Updated. Thanks!

```

You should now have Postgres running in a container in the `openfga` network. However, it will not have the tables required for running OpenFGA. You can use the `migrate` command to create the tables. Using the OpenFGA container, this will look like:
```console

> docker run --rm --network=openfga openfga/openfga migrate --datastore-engine postgres --datastore-uri 'postgres://postgres:password@postgres:5432/postgres?sslmode=disable'

Reviewer: For very long line like this, will it make sense to break it up via `\`?  This allows easier reading.

Author: Sure. Updated. Thanks. How does it look?

```

Finally, start OpenFGA:
```console
docker run --name openfga --network=openfga -p 3000:3000 -p 8080:8080 -p 8081:8081 openfga/openfga run --datastore-engine postgres --datastore-uri 'postgres://postgres:password@postgres:5432/postgres?sslmode=disable'
```

> If all went well, OpenFGA should start and you should see `using 'postgres' storage engine` in the logs.

Reviewer: ```suggestion
OpenFGA should start and you should see `using 'postgres' storage engine` in the logs.
```

Sound more confident in the instruction.

Author: Thanks!

You can choose between two storage options:

> - In-memory storage. This option is not persistent, i.e. if you restart the server, the data will be lost.

Reviewer 3: Can we change this to something like:

OpenFGA supports multiple storage engine options including:

- `memory` - The memory storage engine is the default, but it is not persistent (e.g. data is lost between server restarts).
- `postgres` - The postgres storage engine requires a Postgres instance that the OpenFGA server can reach.

Author: Updated. Thank you!

> - Postgres database. This option requires either a local Postgres installation, or a Docker container that runs Postgres.

Reviewer 2: ```suggestion
- Postgres database. This option requires a running Postgres installation that your OpenFGA server can reach.
```

Whether it is running locally, on docker or on an external network should not be an issue for us.

Author: Ah, good point. Thanks!

> 

Reviewer 2: ### With docker compose

1. Copy the following snippet and paste it into a file. Save it as `docker-compose.yaml`
```yaml
version: '3.8'

networks:
  openfga:

services:
  postgres:
    image: postgres:14
    networks:
      - openfga
    environment:
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=password
      - POSTGRES_DB=openfga
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U postgres" ]
      interval: 5s
      timeout: 5s
      retries: 5

  openfga:
    image: openfga/openfga
    environment:
      - OPENFGA_DATASTORE_ENGINE=postgres
      - OPENFGA_DATASTORE_CONNECTION_URI=postgres://postgres:password@postgres:5432/openfga
      - OPENFGA_LOG_FORMAT=json
    command: run
    depends_on:
      - postgres
    networks:
      - openfga
    ports:
      - "8080:8080"
      - "8081:8081"
      - "3000:3000"
```

2. Run the postgres container in the background

```shell
docker-compose up -d postgres
```

3. Run the necessary OpenFGA migrations

```shell
docker-compose run openfga migrate
```

4. Run the OpenFGA server
```shell
docker-compose up openfga
```

5. Verify that you can reach the openfga server
```shell
curl http://localhost:8080/stores
```

You should see the following output:
```json
{"stores":[], "continuation_token":""}
```

Author: I don't love that a person trying to do this will have to write a new file. As a user, I would find it easier to just copy and paste commands in the terminal. WDYT?

Reviewer 3: I agree with @craigpastro actually. I like it the way it is, because the commands are fully self-contained (no extra file creation/copying necessary).

> #### Configuring Postgres

Reviewer 3: Maybe just "Postgres" for short. The top-level header already says "Configuration Data Storage", so it goes without saying that this is configuration specific content.

Author: Good point. Updated. Thank you!
